src/Widgets/DataFormUploadWidget/DataFormUploadWidgetComponent.tsx

```diff
@@ -40,27 +41,16 @@ provideComponent(DataFormUploadWidget, ({ widget }) => {
     onDropAccepted,
     onDropRejected,
     onDrop: (acceptedFiles) => {
-      setAttachments((prevAttachments) => {
-        const newFiles = acceptedFiles.filter((newFile) => {
-          return !prevAttachments.some(
-            (existingAttachment) =>
-              existingAttachment.file.lastModified === newFile.lastModified &&
-              existingAttachment.file.name === newFile.name &&
-              existingAttachment.file.size === newFile.size &&
-              existingAttachment.file.type === newFile.type,
```

Comment:
* [ ] FYI Thanks to this change only `_id` and `file` stay in use state-wise. The rest of the state properties are now only used to fulfill the interface of rendering `<Attachment />`. Maybe it would be cleaner to get them at render time?